src/controllers/admin/office.ts

```diff
@@ -42,3 +44,155 @@ export const AdminLoginHandler = async (req: Request, res: Response) => {
+      avgregistrations_thisyear = 0;
+    }
+
+    const registrations_alltime = await prismaClient.registration.count();
+
+    const events_alltime = await prismaClient.event.count();
+
+    let avgregistrations_alltime = registrations_alltime/events_alltime;
+
+    if(isNaN(avgregistrations_alltime)){
+      avgregistrations_alltime = 0;
+    }
+
+    const registrations_lastyear = await prismaClient.registration.count({
+      where:{
+        createdAt : {gte : new Date(new Date().getFullYear()-1,0,1), lt : new Date(new Date().getFullYear(),0,1)}
+      }
+    });
+
+    const events_lastyear = await prismaClient.event.count({
+      where:{
+        startTime : {gte : new Date(new Date().getFullYear()-1,0,1), lt : new Date(new Date().getFullYear(),0,1)}
+      }
+    });
+
+    let avgregistrations_lastyear = registrations_lastyear/events_lastyear;
+
+    if(isNaN(avgregistrations_lastyear)){
+      avgregistrations_lastyear = 0;
+    }
+
+    const campaign_thisyear = await prismaClient.campaign.count({
+      where:{
+        createdAt : {gte : new Date(new Date().getFullYear(),0,1)}
+      }
+    })
+
+    const campaign_alltime = await prismaClient.campaign.count();
+
+    const campaign_lastyear = await prismaClient.campaign.count({
+      where:{
+        createdAt : {gte : new Date(new Date().getFullYear()-1,0,1), lt : new Date(new Date().getFullYear(),0,1)}
+      }
+    })
+
+    const data = {
+      ActiveMembers: {
+        ThisYear: activeMembers_thisyear,
+        AllTime: activeMembers_alltime,
+        LastYear: activeMembers_lastyear
+      },
+      EventsCompleted: {
+        ThisYear: EventsCompleted_thisyear,
+        AllTime: EventsCompleted_alltime,
+        LastYear: EventsCompleted_lastyear
+      },
+      AvgAttendeesPerEvent:{
+        ThisYear: avgregistrations_thisyear.toFixed(3),
+        AllTime: avgregistrations_alltime.toFixed(3),
+        LastYear: avgregistrations_lastyear.toFixed(3)
+      },
+      campaigns:{
+        ThisYear: campaign_thisyear,
+        AllTime: campaign_alltime,
+        LastYear: campaign_lastyear
+      }
+    }
+
+    const registration = await prismaClient.event.findMany({
+      select: {
+        startTime: true,
+        Registration:true
+      }
+    })
+
+    const month = ["January","February","March","April","May","June","July","August","September","October","November","December"];
+
+    const attandees_month = new Array(12).fill(0);
+
+    registration.forEach((event)=>{
+      if (event.startTime && event.startTime.getFullYear()==new Date().getFullYear() && event.Registration){
+        attandees_month[new Date(event.startTime).getMonth()] += event.Registration.length;
+      }
+    });
+
+    const charData = attandees_month.map((attendees: number, index: number) => ({
+      month: month[index],attendees: attendees
+    }));
+
+    return res.status(200).json({
+      data,
+      charData
+    });
+  }
+
+
+  catch (e) {
+    return res.status(500).json({
```

Comment:
Log the errors to a file?

Reply:
Actually, logging is going to be a cleanup task after we setup a common Logger class with PinoJS. Putting that one on hold for now. @Ashrockzzz2003 @Nandgopal-R